weblab/entities/views.py

```diff
@@ -778,6 +779,42 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class RenameView(LoginRequiredMixin, UserPassesTestMixin, FormMixin, EntityTypeMixin, DetailView):
+    template_name = 'entities/entity_rename_form.html'
+    context_object_name = 'entity'
+    form_class = EntityRenameForm
+
+    def _get_object(self):
+        if not hasattr(self, 'object'):
+            self.object = self.get_object()
+        return self.object
+
+    def test_func(self):
+        return self._get_object().is_managed_by(self.request.user)
```

Comment:
`is_editable_by` would probably be better than `is_managed_by` here, assuming we want to allow collaborators with write access to rename entities @mirams ?

Reply:
Which means you could just inherit from `EntityCollaboratorRequiredMixin` instead of `UserPassesTestMixin` and remove this method.